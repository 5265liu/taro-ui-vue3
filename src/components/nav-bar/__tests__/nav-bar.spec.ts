import { mount } from '@vue/test-utils'
import { h } from '@vue/runtime-core'
import AtNavBar from '../index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(AtNavBar as any, {
    global: {
      components: {}
    },
    slots,
    props: { ...values },
  })
}

describe('AtNavBar Snap', () => {
  it('render initial AtNavBar', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNavBar -- props class', () => {
    const wrapper = factory({ class: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNavBar -- props style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNavBar -- props fixed', () => {
    const wrapper = factory({ fixed: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNavBar -- props color', () => {
    const wrapper = factory({ color: '#fff' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNavBar -- props leftIconType', () => {
    const wrapper = factory({ leftIconType: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNavBar -- props leftIconType', () => {
    const wrapper = factory({ leftIconType: { value: 'test', color: 'red', size: 36 } })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNavBar -- props leftText', () => {
    const wrapper = factory({ leftText: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNavBar -- props title', () => {
    const wrapper = mount({
      render() {
        return (
          h(AtNavBar, null, {
            default: () => [
              h('text', null, "test")
            ]
          })
        )
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNavBar -- props rightFirstIconType', () => {
    const wrapper = factory({ rightFirstIconType: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNavBar -- props rightFirstIconType', () => {
    const wrapper = factory({ rightFirstIconType: 'test', color: 'red', size: 36 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNavBar -- props rightSecondIconType', () => {
    const wrapper = factory({ rightSecondIconType: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNavBar -- props rightSecondIconType', () => {
    const wrapper = factory({ rightSecondIconType: 'test', color: 'red', size: 36 })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtNavBar Behavior ', () => {
  it('AtNavBar onClickLeftIcon', () => {
    const onClickLeftIcon = jest.fn()
    const wrapper = factory({
      leftIconType: 'test',
      onClickLeftIcon,
    })
    wrapper.find('.at-nav-bar__left-view view').trigger('tap')
    expect(wrapper.element).toMatchSnapshot()
  })

  it('AtNavBar onClickRightSecondIcon', () => {
    const onClickRightSecondIcon = jest.fn()
    const wrapper = factory({
      leftIconType: 'test',
      onClickRightSecondIcon,
    })
    wrapper.find('.at-nav-bar__right-view view:first-child').trigger('tap')
    expect(wrapper.element).toMatchSnapshot()
  })

  it('AtNavBar onClickRightFirstIcon', () => {
    const onClickRightFirstIcon = jest.fn()
    const wrapper = factory({
      leftIconType: 'test',
      onClickRightFirstIcon,
    })
    wrapper.find('.at-nav-bar__right-view view:last-child').trigger('tap')
    expect(wrapper.element).toMatchSnapshot()
  })
})
