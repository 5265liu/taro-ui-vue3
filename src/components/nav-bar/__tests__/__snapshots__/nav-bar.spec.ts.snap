// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtNavBar Behavior  AtNavBar onClickLeftIcon 1`] = `
<view
  class="at-nav-bar"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(97, 144, 232);"
  >
    <text
      class="at-icon at-icon-test"
      style=""
    />
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
  </view>
</view>
`;

exports[`AtNavBar Behavior  AtNavBar onClickRightFirstIcon 1`] = `
<view
  class="at-nav-bar"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(97, 144, 232);"
  >
    <text
      class="at-icon at-icon-test"
      style=""
    />
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
  </view>
</view>
`;

exports[`AtNavBar Behavior  AtNavBar onClickRightSecondIcon 1`] = `
<view
  class="at-nav-bar"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(97, 144, 232);"
  >
    <text
      class="at-icon at-icon-test"
      style=""
    />
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render AtNavBar -- props class 1`] = `
<view
  class="test at-nav-bar test"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(97, 144, 232);"
  >
    
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render AtNavBar -- props color 1`] = `
<view
  class="at-nav-bar"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(255, 255, 255);"
  >
    
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(255, 255, 255);"
    >
      
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(255, 255, 255);"
    >
      
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render AtNavBar -- props fixed 1`] = `
<view
  class="at-nav-bar at-nav-bar--fixed"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(97, 144, 232);"
  >
    
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render AtNavBar -- props leftIconType 1`] = `
<view
  class="at-nav-bar"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(97, 144, 232);"
  >
    <text
      class="at-icon at-icon-test"
      style=""
    />
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render AtNavBar -- props leftIconType 2`] = `
<view
  class="at-nav-bar"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(97, 144, 232);"
  >
    <text
      class="at-icon at-icon-test"
      style="color: red; font-size: 1.536rem;"
    />
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render AtNavBar -- props leftText 1`] = `
<view
  class="at-nav-bar"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(97, 144, 232);"
  >
    
    <text
      class="at-nav-bar__text"
    >
      test
    </text>
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render AtNavBar -- props rightFirstIconType 1`] = `
<view
  class="at-nav-bar"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(97, 144, 232);"
  >
    
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
    <view
      class="at-nav-bar__container"
      style="color: rgb(97, 144, 232);"
    >
      <text
        class="at-icon at-icon-test"
        style=""
      />
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render AtNavBar -- props rightFirstIconType 2`] = `
<view
  class="at-nav-bar"
  size="36"
>
  <view
    class="at-nav-bar__left-view"
    style="color: red;"
  >
    
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: red;"
    >
      
    </view>
    <view
      class="at-nav-bar__container"
      style="color: red;"
    >
      <text
        class="at-icon at-icon-test"
        style=""
      />
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render AtNavBar -- props rightSecondIconType 1`] = `
<view
  class="at-nav-bar"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(97, 144, 232);"
  >
    
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container"
      style="color: rgb(97, 144, 232);"
    >
      <text
        class="at-icon at-icon-test"
        style=""
      />
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render AtNavBar -- props rightSecondIconType 2`] = `
<view
  class="at-nav-bar"
  size="36"
>
  <view
    class="at-nav-bar__left-view"
    style="color: red;"
  >
    
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container"
      style="color: red;"
    >
      <text
        class="at-icon at-icon-test"
        style=""
      />
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: red;"
    >
      
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render AtNavBar -- props style 1`] = `
<view
  class="at-nav-bar"
  style="color: red;"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(97, 144, 232);"
  >
    
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render AtNavBar -- title as both slot and prop 1`] = `
<view>
  <view
    class="at-nav-bar"
  >
    <view
      class="at-nav-bar__left-view"
      style="color: rgb(97, 144, 232);"
    >
      
      <text
        class="at-nav-bar__text"
      />
    </view>
    <view
      class="at-nav-bar__title"
    >
      <text>
        title as slot
      </text>
    </view>
    <view
      class="at-nav-bar__right-view"
    >
      <view
        class="at-nav-bar__container at-nav-bar__container--hide"
        style="color: rgb(97, 144, 232);"
      >
        
      </view>
      <view
        class="at-nav-bar__container at-nav-bar__container--hide"
        style="color: rgb(97, 144, 232);"
      >
        
      </view>
    </view>
  </view>
  <view
    class="at-nav-bar"
  >
    <view
      class="at-nav-bar__left-view"
      style="color: rgb(97, 144, 232);"
    >
      
      <text
        class="at-nav-bar__text"
      />
    </view>
    <view
      class="at-nav-bar__title"
    >
      title as prop
    </view>
    <view
      class="at-nav-bar__right-view"
    >
      <view
        class="at-nav-bar__container at-nav-bar__container--hide"
        style="color: rgb(97, 144, 232);"
      >
        
      </view>
      <view
        class="at-nav-bar__container at-nav-bar__container--hide"
        style="color: rgb(97, 144, 232);"
      >
        
      </view>
    </view>
  </view>
</view>
`;

exports[`AtNavBar Snap render initial AtNavBar 1`] = `
<view
  class="at-nav-bar"
>
  <view
    class="at-nav-bar__left-view"
    style="color: rgb(97, 144, 232);"
  >
    
    <text
      class="at-nav-bar__text"
    />
  </view>
  <view
    class="at-nav-bar__title"
  />
  <view
    class="at-nav-bar__right-view"
  >
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
    <view
      class="at-nav-bar__container at-nav-bar__container--hide"
      style="color: rgb(97, 144, 232);"
    >
      
    </view>
  </view>
</view>
`;
